packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -3579,4 +3588,30 @@ export class SnapController extends BaseController<
       },
     });
   }
+
+  /**
+   * Calculate change of initialConnections.
+   *
+   * @param oldConnectionsSet - Previously approved connections.
+   * @param desiredConnectionsSet - New connections.
+   * @returns Object containing newConnections, unusedConnections and approvedConnections.
+   */
+  #calculateConnectionsChange(
```

Comment:
I think this should be a utility function instead of a private class method since it doesn't use any class properties

Reply:
Refactored. Moved to utils.